packages/smithy-client/src/command.spec.ts

```diff
@@ -1,6 +1,27 @@
 import { Command } from "./command";
 
 describe(Command.name, () => {
+  it("has optional argument if the input type has no required members", async () => {
+    type OptionalInput = {
+      key?: string;
+      optional?: string;
+    };
+
+    type RequiredInput = {
+      key: string | undefined;
+      optional?: string;
+    };
+
+    class WithRequiredInputCommand extends Command.classBuilder<RequiredInput, any, any, any, any>()
+      .build() {}
+
+    class WithOptionalInputCommand extends Command.classBuilder<OptionalInput, any, any, any, any>()
+      .build() {}
+
+    new WithRequiredInputCommand({ key: "1" });
+
+    new WithOptionalInputCommand(); // expect no type error.
```

Comment:
Is there a way to test the following cases?

- `WithRequiredInputCommand` without an input => emits a type error
- `WithOptionalInputCommand` with an input => emits no type error

Reply:
can't test a type _error_ because the test won't compile to run, but I'll add a typescript compiler test